### PR TITLE
add agent.train()/eval()

### PR DESCRIPTION
**Pull request by ShuaibinLi**
## Add agent.train(), agent.eval()
model/alg._train()/_eval()

agent.train()/eval()

### CODE REVIEW

File: parl/core/paddle/agent.py [hunk 1]
```diff
@@ -180,3 +183,18 @@ def restore(self, save_path, model=None):
             model = self.alg.model
         param_dict = paddle.load(save_path)
         model.set_state_dict(param_dict)
+
+    def train(self):
+        """Sets the agent in training mode.
+
+        This has any effect only on certain modules of their behaviors in training/evaluation mode, if they are affected, e.g. class:`Dropout`, class:`BatchNorm`, etc.
```
Comment (rical730) [inline parl/core/paddle/agent.py]: This has any effect only on certain modules, which behave differently in train/evaluation mode, eg. `Dropout`, `BatchNorm`, etc.

File: parl/core/paddle/agent.py [hunk 2]
```diff
@@ -65,6 +67,7 @@ def __init__(self, algorithm):
 
         assert isinstance(algorithm, Algorithm)
         super(Agent, self).__init__(algorithm)
+        self.training = True
```
Comment (rical730) [inline parl/core/paddle/agent.py:71]: Please provide comment for self.training

File: parl/core/torch/agent.py [hunk 3]
```diff
@@ -147,3 +150,18 @@ def restore(self, save_path, model=None, map_location=None):
             model = self.alg.model
         checkpoint = torch.load(save_path, map_location=map_location)
         model.load_state_dict(checkpoint)
+
+    def train(self):
+        """Sets the agent in training mode.
+
+        This has any effect only on certain modules of their behaviors in training/evaluation mode, if they are affected, e.g. class:`Dropout`, class:`BatchNorm`, etc.
```
Comment (rical730) [inline parl/core/torch/agent.py]: This has any effect only on certain modules, which behave differently in train/evaluation mode, eg. `Dropout`, `BatchNorm`, etc.

File: parl/core/torch/agent.py [hunk 4]
```diff
@@ -70,6 +72,7 @@ def __init__(self, algorithm):
 
         assert isinstance(algorithm, Algorithm)
         super(Agent, self).__init__(algorithm)
+        self.training = True
```
Comment (rical730) [inline parl/core/torch/agent.py:76]: Please provide comment for self.training

File: parl/core/paddle/agent.py (hunk 1 above)
Comment (ShuaibinLi) [reply parl/core/paddle/agent.py]: done

**Review by rical730 — APPROVED** [review]
LGTM~
